neuralop/layers/discrete_continuous_convolution.py

```diff
@@ -0,0 +1,611 @@
+            assert len(n_out) == 2
+            x, wx = _precompute_grid(n_out[0], grid=grid_out, periodic=periodic)
+            y, wy = _precompute_grid(n_out[1], grid=grid_out, periodic=periodic)
+            x, y = torch.meshgrid(torch.from_numpy(x), torch.from_numpy(y))
+            grid_out = torch.stack([x.reshape(-1), y.reshape(-1)])
+        else:
+            raise ValueError(f"Unknown grid output type of type {type(grid_out)}")
+
+        # check that input arrays are valid point clouds in 2D
+        assert len(grid_in.shape) == 2
+        assert len(grid_out.shape) == 2
+        assert len(quad_weights.shape) == 1
+        assert grid_in.shape[0] == 2
+        assert grid_out.shape[0] == 2
+
+        self.n_in = grid_in.shape[-1]
+        self.n_out = grid_out.shape[-1]
+
+        # compute the cutoff radius based on the bandlimit of the input field
+        # TODO: Attention - this heuristic is ad-hoc! Make sure to set it yourself!
+        if radius_cutoff is None:
+            radius_cutoff = radius_cutoff = 2 / float(math.sqrt(self.n_out) - 1)
+
+        if radius_cutoff <= 0.0:
+            raise ValueError("Error, radius_cutoff has to be positive.")
+
+        # integration weights
+        self.register_buffer("quad_weights", quad_weights, persistent=False)
+
+        idx, vals = _precompute_convolution_tensor_2d(grid_in, grid_out, self.kernel_shape, quad_weights, radius_cutoff=radius_cutoff, periodic=periodic)
+
+        # to improve performance, we make psi a matrix by merging the first two dimensions
+        # This has to be accounted for in the forward pass
+        idx = torch.stack([idx[0] * self.n_out + idx[1], idx[2]], dim=0)
+
+        self.register_buffer("psi_idx", idx.contiguous(), persistent=False)
+        self.register_buffer("psi_vals", vals.contiguous(), persistent=False)
+
+    def get_psi(self):
+        psi = torch.sparse_coo_tensor(self.psi_idx, self.psi_vals, size=(self.kernel_size * self.n_out, self.n_in))
+        return psi
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        # pre-multiply x with the quadrature weights
+        x = self.quad_weights * x
+
+        psi = self.get_psi()
+
+        # extract shape
+        B, C, _ = x.shape
+
+        # bring into the right shape for the bmm and perform it
+        x = x.reshape(B * C, self.n_in).permute(1, 0).contiguous()
+        x = torch.mm(psi, x)
+        x = x.permute(1, 0).reshape(B, C, self.kernel_size, self.n_out)
+        x = x.reshape(B, self.groups, self.groupsize, self.kernel_size, self.n_out)
+
+        # do weight multiplication
+        out = torch.einsum("bgckx,gock->bgox", x, self.weight.reshape(self.groups, -1, self.weight.shape[1], self.weight.shape[2]))
+        out = out.reshape(out.shape[0], -1, out.shape[-1])
+
+        if self.bias is not None:
+            out = out + self.bias.reshape(1, -1, 1)
+
+        return out
+
+
+class DiscreteContinuousConvTranspose2d(DiscreteContinuousConv):
```

Comment:
could we replace all instances of `DiscreteContinuous` with DISCO? The names are quite wordy.

Reply:
I think this is a matter of taste. How about creating aliases for those? The reasoning is that writing it out helps when printing model.

Reply:
As per earlier conversation, I agree we should keep the long names